### PR TITLE
draft: bert cp support

### PR DESCRIPTION
**Pull request by JimmyZhang12**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/models/language_modeling/megatron_bert_model.py [hunk 1]
```diff
@@ -567,6 +575,39 @@ def on_test_epoch_end(self):
...
+        for key, val in batch.items():
+            assert len(val.shape) <= 2
+            if len(val.shape) == 2:
+                batch[key] = batch[key][..., seq_start_idx:seq_end_idx]
+
+        return batch
+
+    def get_batch(self, dataloader_iter):
+        batch, batch_idx, dataloader_idx = next(dataloader_iter)
+        batch['padding_mask'] = torch.ones_like(batch['padding_mask'])
```
[the remark this answers is not in the document]
Comment (JimmyZhang12) [reply nemo/collections/nlp/models/language_modeling/megatron_bert_model.py:607]: yes- I had left this in from debugging, will remove.

File: nemo/collections/nlp/models/language_modeling/megatron/bert/bert_spec.py [hunk 2]
```diff
@@ -41,52 +41,70 @@
...
-            submodules=SelfAttentionSubmodules(
-                linear_qkv=TEColumnParallelLinear,
-                core_attention=TEDotProductAttention,
-                linear_proj=TERowParallelLinear,
-                q_layernorm=IdentityOp,
-                k_layernorm=IdentityOp,
+def get_bert_layer_with_transformer_engine_spec_postln(cp_size):
+    # Transformer Engine currently only supports 'no_mask' with CP for non-causal models
+    # attn_mask_type = AttnMaskType.no_mask if cp_size > 1 else AttnMaskType.padding
+    attn_mask_type = AttnMaskType.no_mask
```
[the remark this answers is not in the document]
Comment (JimmyZhang12) [reply nemo/collections/nlp/models/language_modeling/megatron/bert/bert_spec.py:49]: Yes, I accidentally left this in from debugging, line 48 should be uncommented, will remove
